.travis.yml

```diff
@@ -50,8 +50,8 @@ matrix:
         # value so the tests pass. The real environment variable will be checked during deployment.
         - docker-compose run -e SITE_URL=https://treeherder.dev -e TREEHERDER_DEBUG=False backend python -bb ./manage.py check --deploy --fail-level WARNING
         # Using `-bb` mode to surface BytesWarnings: https://docs.python.org/3.7/using/cmdline.html#cmdoption-b
-        - docker-compose run backend python -bb -m pytest tests/ --runslow --ignore=tests/selenium/
-
+        - docker-compose run backend bash ./tests/coverage.sh
```

Comment:
I believe you don't need to use `bash` in that line. Let me know if I'm wrong.

Reply:
I want to take advantage of `exec "$@"` in the coverage script.

You can then do this:
```
docker-compose run backend ./tests/coverage.sh ./runtests.sh
docker-compose run backend ./tests/coverage.sh pytest tests/ --runslow --ignore=tests=selenium/
docker-compose run backend ./tests/coverage.sh python -bb -m pytest tests/selenium/
```

Reply:
Instead of taking advantage of the `exec` trick and the script. Use `after_success: pytest --cov=./ --cov-report=xml && codecov -t 995256be-3a1f-4d10-b181-e24f88603b6b`

Reply:
> I believe you don't need to use `bash` in that line. Let me know if I'm wrong.

Sir, if we don't use bash, ./docker/entrypoint.sh is denying the permission to execute

Reply:
> Instead of taking advantage of the `exec` trick and the script. Use `after_success: pytest --cov=./ --cov-report=xml && codecov -t 995256be-3a1f-4d10-b181-e24f88603b6b`

The issue with this is that there are two docker instances. One for Travis and one for our backend. Now the issue with after success is that it will try to find the xml file in the travis docker file, which it won't find. That's why, I tried to run and push the tests coverage in the inner docker container itself.

Reply:
Oh dear. This is why we need the GSoC project.

Use the `exec` approach I first mentioned.

Reply:
> I want to take advantage of `exec "$@"` in the coverage script.
> 
> You can then do this:
> 
> ```
> docker-compose run backend ./tests/coverage.sh ./runtests.sh
> docker-compose run backend ./tests/coverage.sh pytest tests/ --runslow --ignore=tests=selenium/
> docker-compose run backend ./tests/coverage.sh python -bb -m pytest tests/selenium/
> ```

@armenzg I am not able to understand the outcome goal here. Can you please brief this a little more?

Reply:
You will need to change the following lines:
https://github.com/mozilla/treeherder/blob/master/.travis.yml#L41
https://github.com/mozilla/treeherder/blob/master/.travis.yml#L53
https://github.com/mozilla/treeherder/blob/master/.travis.yml#L71

We run the Python tests in 3 batches and all will have to generate and upload coverage information.

Reply:
> You will need to change the following lines:
> https://github.com/mozilla/treeherder/blob/master/.travis.yml#L41
> https://github.com/mozilla/treeherder/blob/master/.travis.yml#L53
> https://github.com/mozilla/treeherder/blob/master/.travis.yml#L71
> 
> We run the Python tests in 3 batches and all will have to generate and upload coverage information.

Understood. Need a standard coverage file which should be passed as a parameter to every test that we are conducting. For that, need to use ```exec```